### PR TITLE
Update vendored dependencies to released version

### PR DESCRIPTION
**Pull request by dims**
- github.com/jteeuwen/go-bindata is no longer active
  So we should move to the newer home (github.com/go-bindata/go-bindata)
- Updating to github.com/go-bindata/go-bindata v3.1.1+incompatible

Updating a bunch of dependencies to released/tagged versions:
- node:
  - [x] github.com/opencontainers/go-digest v1.0.0-rc1 - @mattjmcnaughton @BenTheElder 
  - [x] github.com/opencontainers/image-spec v1.0.1 - @liggitt
  - [x] github.com/opencontainers/selinux v1.2.2 -  @BenTheElder 
  - [x] github.com/mistifyio/go-zfs v2.1.1+incompatible - @BenTheElder 
  - [ ] ~~github.com/seccomp/libseccomp-golang v0.9.0~~ (DROP-ing this - see reviews below)
  - [x] github.com/cyphar/filepath-securejoin v0.2.2 - @mattjmcnaughton @BenTheElder 
  - [x] github.com/godbus/dbus v4.1.0+incompatible -  @BenTheElder
- windows, network
  - [x] github.com/miekg/dns v1.1.3 - @johnbelamaric
- cloud-provider
  - ~~github.com/d2g/dhcp4client v1.0.0~~ can drop because https://github.com/kubernetes/kubernetes/pull/72178 merged (DONE)
  - ~~github.com/xanzy/go-cloudstack v2.4.1+incompatible~~ can drop because https://github.com/kubernetes/kubernetes/pull/72178 merged (DONE)
- kubeadm
  - [x] github.com/mholt/caddy v1.0.0 - @neolit123 and @johnbelamaric
- apimachinery:
  - [x] github.com/coreos/go-semver v0.3.0 - @liggitt
  - [x] github.com/emicklei/go-restful v2.9.5+incompatible - @liggitt
  - [x] github.com/evanphx/json-patch v4.2.0+incompatible - @liggitt
  - [x] gopkg.in/yaml.v2 v2.2.2 - @liggitt
  - [x] github.com/json-iterator/go v1.1.6 - @liggitt
- auth:
  - [x] gopkg.in/square/go-jose.v2 v2.2.2 - @liggitt
- cli:
  - [x] github.com/spf13/afero v1.2.2 - @liggitt
  - [x] github.com/spf13/cast v1.3.0 - @BenTheElder
  - [x] github.com/spf13/cobra v0.0.4 - @liggitt
  - [x] github.com/spf13/jwalterweatherman v1.1.0 - @liggitt: unfortunate init block
  - [x] github.com/spf13/pflag v1.0.3 - @liggitt
  - [x] github.com/spf13/viper v1.3.2 - @liggitt
  - [x] github.com/magiconair/properties v1.8.1 - @liggitt
  - [x] github.com/mitchellh/go-wordwrap v1.0.0 - @liggitt
  - [x] github.com/russross/blackfriday v1.5.2 - @liggitt
  - [x] github.com/cpuguy83/go-md2man v1.0.10 - @liggitt
  - [x] github.com/fatih/camelcase v1.0.0 - @liggitt
- apps / cronjob
  - [x] github.com/robfig/cron v1.1.0 - @liggitt
- trivial or test-only bumps:
  - [x] github.com/mattn/go-shellwords v1.0.5 - @liggitt
  - [x] github.com/kr/pretty v0.1.0 - @liggitt
  - [x] github.com/mvdan/xurls v1.1.0 - @liggitt
  - [x] github.com/jonboulle/clockwork v0.1.0 - @liggitt
  - [x] github.com/client9/misspell v0.3.4 - @liggitt
  - [x] github.com/go-bindata/go-bindata v3.1.1+incompatible - @liggitt
  - [x] github.com/golang/mock v1.2.0 - @liggitt
  - [x] github.com/google/gofuzz v1.0.0 - @liggitt
  - [x] github.com/onsi/ginkgo v1.8.0 - @liggitt
  - [x] github.com/onsi/gomega v1.5.0 - @liggitt
  - [x] github.com/stretchr/testify v1.3.0 - @liggitt

Deleted the dependencies no longer used:
github.com/hashicorp/hcl
github.com/jteeuwen/go-bindata
github.com/kr/fs
github.com/kr/text
github.com/pelletier/go-toml
github.com/pkg/sftp
github.com/shurcooL/sanitized_anchor_name
github.com/spf13/cast

Change-Id: I63674d0fbfc41c51993276b73e2b2bd056c064ec

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Related to: https://github.com/kubernetes/kubernetes/issues/78434

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by liggitt** [comment]
I know it's a pain, but can we split these into separate commits (or at least group related ones into distinct commits)? bumping ~50 deps in a single commit is asking for a mega-revert :-/

for review purposes, it might be good to target these by owning/interested sig, e.g.:
* sig-testing (ginkgo, gomega, testify, golang/mock)
* sig-node (dbus, opencontainers, seccomp)
* sig-instrumentation (prometheus)
* sig-api-machinery (go-restful, json-patch, protobuf, yaml.v2)
* sig-auth (go-jose)
* sig-cli (spf13)

the golang.org/x/... are currently on the go 1.12 release branches, was there a reason you bumped those?

**Comment by dims** [comment]
@liggitt ack. will chop it up.

**Comment by dims** [comment]
automated using script https://pasted.tech/pastes/65057a26fc8cc68f45d8e4b55356f63e465e8145.raw :)

**Comment by liggitt** [comment]
> automated using script https://pasted.tech/pastes/65057a26fc8cc68f45d8e4b55356f63e465e8145.raw :)

shhhh, don't tell people how easy it is... you could have claimed three days of work and no one would have guessed

File: staging/src/k8s.io/api/go.mod
Comment (liggitt) [inline staging/src/k8s.io/api/go.mod]: hmm... seeing a lot of new replace statements in staging go.mod files, which is not good

**Comment by dims** [comment]
@liggitt the x/net is being pulled by miekg/dns (see https://github.com/dims/kubernetes/commit/7252d1c65d79a05293163af5797f20e6c0188b68)

**Comment by dims** [comment]
@liggitt Here's what's left that i will *not* be trying to add to this PR

github.com/Microsoft/hcsshim
github.com/docker/distribution
github.com/docker/docker
github.com/docker/libnetwork
github.com/google/cadvisor
github.com/googleapis/gnostic
github.com/opencontainers/runc
github.com/vishvananda/netlink
google.golang.org/api

Each of them has some problem or another.  We can deal with them later. This PR has only mechanical changes that i could script. The exception is the `go-bindata/go-bindata` which still was using sed/mv only.

**Comment by liggitt** [comment]
this introduces a lot of deltas between pinned and preferred versions:

```
hack/lint-dependencies.sh 
These modules are pinned to versions different than the minimal preferred version.
That means that without require directives, a different version would be selected.
The command to switch to the minimal preferred version is listed for each module.

github.com/BurntSushi/toml
    pinned:    v0.3.0
    preferred: v0.3.1
    hack/pin-dependency.sh github.com/BurntSushi/toml v0.3.1
github.com/gogo/protobuf
    pinned:    v0.0.0-20171007142547-342cbe0a0415
    preferred: v1.1.1
    hack/pin-dependency.sh github.com/gogo/protobuf v1.1.1
github.com/google/uuid
    pinned:    v1.0.0
    preferred: v1.1.1
    hack/pin-dependency.sh github.com/google/uuid v1.1.1
github.com/gorilla/websocket
    pinned:    v0.0.0-20170926233335-4201258b820c
    preferred: v1.4.0
    hack/pin-dependency.sh github.com/gorilla/websocket v1.4.0
github.com/kr/text
    pinned:    v0.0.0-20130911015532-6807e777504f
    preferred: v0.1.0
    hack/pin-dependency.sh github.com/kr/text v0.1.0
github.com/onsi/ginkgo
    pinned:    v1.6.0
    preferred: v1.8.0
    hack/pin-dependency.sh github.com/onsi/ginkgo v1.8.0
github.com/spf13/viper
    pinned:    v0.0.0-20160820190039-7fb2782df3d8
    preferred: v1.3.2
    hack/pin-dependency.sh github.com/spf13/viper v1.3.2
... expected golang.org/x/... ones...
gopkg.in/natefinch/lumberjack.v2
    pinned:    v2.0.0-20150622162204-20b71e5b60d7
    preferred: v2.0.0
    hack/pin-dependency.sh gopkg.in/natefinch/lumberjack.v2 v2.0.0

Pinned module versions that aren't actually used:
GO111MODULE=on go mod edit -dropreplace github.com/jteeuwen/go-bindata
GO111MODULE=on go mod edit -dropreplace github.com/kr/fs
GO111MODULE=on go mod edit -dropreplace github.com/pkg/sftp
GO111MODULE=on go mod edit -dropreplace github.com/shurcooL/sanitized_anchor_name
```

**Comment by dims** [comment]
@liggitt leave the following alone. protobuf, ginkgo, cast all of them cause some problems or another.

```
github.com/gogo/protobuf
    pinned:    v0.0.0-20171007142547-342cbe0a0415
    preferred: v1.1.1
    hack/pin-dependency.sh github.com/gogo/protobuf v1.1.1
github.com/onsi/ginkgo
    pinned:    v1.6.0
    preferred: v1.8.0
    hack/pin-dependency.sh github.com/onsi/ginkgo v1.8.0
github.com/spf13/cast
    pinned:    v1.0.0
    preferred: v1.3.0
    hack/pin-dependency.sh github.com/spf13/cast v1.3.0
golang.org/x/crypto
    pinned:    v0.0.0-20181025213731-e84da0312774
    preferred: v0.0.0-20190228161510-8dd112bcdc25
    hack/pin-dependency.sh golang.org/x/crypto v0.0.0-20190228161510-8dd112bcdc25
golang.org/x/net
    pinned:    v0.0.0-20190206173232-65e2d4e15006
    preferred: v0.0.0-20190328230028-74de082e2cca
    hack/pin-dependency.sh golang.org/x/net v0.0.0-20190328230028-74de082e2cca
golang.org/x/sync
    pinned:    v0.0.0-20181108010431-42b317875d0f
    preferred: v0.0.0-20181221193216-37e7f081c4d4
    hack/pin-dependency.sh golang.org/x/sync v0.0.0-20181221193216-37e7f081c4d4
golang.org/x/sys
    pinned:    v0.0.0-20190209173611-3b5209105503
    preferred: v0.0.0-20190312061237-fead79001313
    hack/pin-dependency.sh golang.org/x/sys v0.0.0-20190312061237-fead79001313
golang.org/x/tools
    pinned:    v0.0.0-20190313210603-aa82965741a9
    preferred: v0.0.0-20190328211700-ab21143f2384
    hack/pin-dependency.sh golang.org/x/tools v0.0.0-20190328211700-ab21143f2384
```

**Comment by liggitt** [comment]
> * protobuf : needs new code to be generated

leaving the staging repos preferring a version of gogo/protobuf that does not match their generated proto/pb files is not workable. we either need to:
1. bump the proto library and regenerate the proto in this PR (I'm adding fixtures in https://github.com/kubernetes/kubernetes/pull/78309 so we can verify a PR updating proto libraries doesn't break encoding/decoding)
2. defer bumping the dependencies that pull in the proto 1.1.1 bump

**Comment by dims** [comment]
ok, so we now are left with these

```
[dims@dims-mac 17:20] ~/go/src/k8s.io/kubernetes ⟩ hack/lint-dependencies.sh
These modules are pinned to versions different than the minimal preferred version.
That means that without require directives, a different version would be selected.
The command to switch to the minimal preferred version is listed for each module.

github.com/onsi/ginkgo
    pinned:    v1.6.0
    preferred: v1.8.0
    hack/pin-dependency.sh github.com/onsi/ginkgo v1.8.0
golang.org/x/crypto
    pinned:    v0.0.0-20181025213731-e84da0312774
    preferred: v0.0.0-20190228161510-8dd112bcdc25
    hack/pin-dependency.sh golang.org/x/crypto v0.0.0-20190228161510-8dd112bcdc25
golang.org/x/net
    pinned:    v0.0.0-20190206173232-65e2d4e15006
    preferred: v0.0.0-20190328230028-74de082e2cca
    hack/pin-dependency.sh golang.org/x/net v0.0.0-20190328230028-74de082e2cca
golang.org/x/sync
    pinned:    v0.0.0-20181108010431-42b317875d0f
    preferred: v0.0.0-20181221193216-37e7f081c4d4
    hack/pin-dependency.sh golang.org/x/sync v0.0.0-20181221193216-37e7f081c4d4
golang.org/x/sys
    pinned:    v0.0.0-20190209173611-3b5209105503
    preferred: v0.0.0-20190312061237-fead79001313
    hack/pin-dependency.sh golang.org/x/sys v0.0.0-20190312061237-fead79001313
golang.org/x/tools
    pinned:    v0.0.0-20190313210603-aa82965741a9
    preferred: v0.0.0-20190328211700-ab21143f2384
    hack/pin-dependency.sh golang.org/x/tools v0.0.0-20190328211700-ab21143f2384
```

**Comment by dims** [comment]
fyi, updating github.com/mholt/caddy to v1.0.0 drags in github.com/onsi/ginkgo v1.8.0

**Comment by dims** [comment]
found a fix for the viper issue. we are now down to the following output from lint-dependencies


```
golang.org/x/crypto
    pinned:    v0.0.0-20181025213731-e84da0312774
    preferred: v0.0.0-20190228161510-8dd112bcdc25
    hack/pin-dependency.sh golang.org/x/crypto v0.0.0-20190228161510-8dd112bcdc25
golang.org/x/net
    pinned:    v0.0.0-20190206173232-65e2d4e15006
    preferred: v0.0.0-20190328230028-74de082e2cca
    hack/pin-dependency.sh golang.org/x/net v0.0.0-20190328230028-74de082e2cca
golang.org/x/sync
    pinned:    v0.0.0-20181108010431-42b317875d0f
    preferred: v0.0.0-20181221193216-37e7f081c4d4
    hack/pin-dependency.sh golang.org/x/sync v0.0.0-20181221193216-37e7f081c4d4
golang.org/x/sys
    pinned:    v0.0.0-20190209173611-3b5209105503
    preferred: v0.0.0-20190312061237-fead79001313
    hack/pin-dependency.sh golang.org/x/sys v0.0.0-20190312061237-fead79001313
golang.org/x/tools
    pinned:    v0.0.0-20190313210603-aa82965741a9
    preferred: v0.0.0-20190328211700-ab21143f2384
    hack/pin-dependency.sh golang.org/x/tools v0.0.0-20190328211700-ab21143f2384
```

**Comment by dims** [comment]
/retest

-- dims

**Comment by dims** [comment]
@cblecker @liggitt - this is ready now.

Other than the golang.org/x/* repos are the only ones remaining when i run `hack/lint-dependencies`. the viper one had a small code change, they added a new error when config files are missing.

**Comment by BenTheElder** [comment]
Not sure how to approach reviewing this, but I do think we should avoid leaving our dependencies stagnant..

**Comment by liggitt** [comment]
> Not sure how to approach reviewing this, but I do think we should avoid leaving our dependencies stagnant..

I'm also in favor in general.

Thanks for pulling out the proto library bumps. Once we have compatibility testing around proto encoding/decoding (in progress in https://github.com/kubernetes/kubernetes/pull/78309) I'll be much more comfortable bumping our proto generation libraries.

Two questions on timing:
* @kubernetes/sig-scalability-pr-reviews - I know past bumps like this have been bisect targets for scalability investigations. Are scale tests/targets currently in a good spot on master or are we still investigating regressions?
* I'm giving priority to reviews for things targeting code freeze. Are there urgent bumps in here that must make it into 1.15 or is this just general hygiene?

**Comment by wojtek-t** [comment]
> @kubernetes/sig-scalability-pr-reviews - I know past bumps like this have been bisect targets for scalability investigations. Are scale tests/targets currently in a good spot on master or are we still investigating regressions?

We're finally in reasonable shape - last 6 runs (so week+) is fully green:
https://testgrid.k8s.io/sig-scalability-gce#gce-scale-performance
So yes - we're in good point to spot the potential problems. (though it all depends on how many other changes are merging in the meantime - last days before code freeze are always risky...)

**Review by cblecker — APPROVED** [review]
If @liggitt is cool with it, then I am too. We can revert things if needed.

**Comment by wojtek-t** [comment]
I'm fine with trying with a caveat that in case of regression I reserve a right to revert.

**Comment by liggitt** [comment]
were the changes to things in our core functional path (go-restful, json-patch, json-iterator, yaml.v2, etc) been reviewed? I'm in favor of keeping up to date, but not of blindly bumping levels.

**Comment by liggitt** [comment]
talked with @dims offline, will look to get this in first thing in 1.16

/milestone v1.16

**Comment by liggitt** [comment]
can we drop the `golang.org/x/crypto v0.0.0-20190228161510-8dd112bcdc25` change and stay on the go-1.12 branch `e84da0312774`?

File: vendor/github.com/spf13/jwalterweatherman/default_notepad.go
Comment (liggitt) [inline vendor/github.com/spf13/jwalterweatherman/default_notepad.go]: this is unfortunate this is happening in an init() block

Comment (liggitt) [reply vendor/github.com/spf13/jwalterweatherman/default_notepad.go]: still not a fan, but verified this doesn't hit disk

**Comment by liggitt** [comment]
* needs a rebase, and can drop the cloudstack and dhcp4client commits and deps now that https://github.com/kubernetes/kubernetes/pull/72178 merged
* squash the single-line "dropping ..." commits into a single commit
* if possible, squash or drop the churning commits (prometheus... rollback prometheus, viper 1.0.0->1.3.2->revert 1.3.2, etc)
* get the named sigs to review the bumps of the unchecked dependencies (and ack which ones they reviewed in comments)

once the remaining deps are reviewed, this lgtm

**Comment by dims** [comment]
/assign @bclau 
(for sig windows)

/assign @yujuhong @resouer 
(for sig node)

**Comment by neolit123** [comment]
> kubeadm
> github.com/mholt/caddy v1.0.0

LGTM. it's on my TODO to remove this dep.

this move seems a bit odd:
>  vendor/github.com/spf13/afero/sftp/BUILD → vendor/github.com/mholt/caddy/telemetry/BUILD

**Comment by liggitt** [comment]
> this move seems a bit odd:
> 
> > vendor/github.com/spf13/afero/sftp/BUILD → vendor/github.com/mholt/caddy/telemetry/BUILD

that's just git file move detection getting confused by the removal and addition of substantially similar files, not an issue

**Comment by bclau** [comment]
> 
> 
> > this move seems a bit odd:
> > > vendor/github.com/spf13/afero/sftp/BUILD → vendor/github.com/mholt/caddy/telemetry/BUILD
> 
> that's just git file move detection getting confused by the removal and addition of substantially similar files, not an issue

True, I've seen it a couple of times today as well.

FWIW, it seems that only the telemetry part is new for caddy, so I don't think regressions are likely with the caddy update.

File: go.mod
Comment (andrewsykim) [inline go.mod]: Any reason to bump this if it's already a tagged release?

Comment (dims) [reply go.mod]: One of the other dependencies needed this update @andrewsykim

Comment (andrewsykim) [reply go.mod]: ah okay, thank you

File: vendor/github.com/onsi/ginkgo/ginkgo/run_watch_and_build_command_flags.go
Comment (andrewsykim) [inline vendor/github.com/onsi/ginkgo/ginkgo/run_watch_and_build_command_flags.go]: Wonder if there's anything useful to get out of this new feature

**Comment by dims** [comment]
/hold

(for reviews to complete)

File: go.mod
Comment (johnbelamaric) [inline go.mod]: The original of miekg/dns was from June 2016, which actually pre-dates 1.0. I don't see any reason to pick 1.1.3, we should just go to the latest which is 1.1.14.

File: go.mod
Comment (johnbelamaric) [inline go.mod]: this is ok

**Comment by mattjmcnaughton** [comment]
@dims thanks for doing this :)

Would you mind writing a brief description of the process for "reviewing" a vendored dep update such as this one? I'd love to help, but I'm afraid I don't have the context/knowledge to be useful.

**Comment by dims** [comment]
@mattjmcnaughton basically look through the diffs to see if there are any red flags, one example from Jordan's review earlier was someone had added a "init()" method (which was not too bad). 
Let's take an example `https://github.com/spf13/cast` we are stating from SHA `e31f36ffc91a` and updating to `v1.3.0`. So it's easy to compose a github URL to compare these two points namely - `https://github.com/spf13/cast/compare/e31f36ffc91a...v1.3.0` [compare url](https://github.com/spf13/cast/compare/e31f36ffc91a...v1.3.0) If you hit the url, you will see the list of changes and the files changed as well. You can also look at who is calling the `spf13/cast` (my favorite is using [rip-grep](https://github.com/BurntSushi/ripgrep))  and you will see that we are not calling it directly from k/k but is used from other code in the vendor directory. It would also be easy to see exactly which methods are being called if you want to go deeper. 

You should also see the other things mentioned here:
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/godep.md#reviewing-and-approving-dependency-changes

**Comment by mattjmcnaughton** [comment]
> @mattjmcnaughton basically look through the diffs to see if there are any red flags, one example from Jordan's review earlier was someone had added a "init()" method (which was not too bad).
> Let's take an example `https://github.com/spf13/cast` we are stating from SHA `e31f36ffc91a` and updating to `v1.3.0`. So it's easy to compose a github URL to compare these two points namely - `https://github.com/spf13/cast/compare/e31f36ffc91a...v1.3.0` [compare url](https://github.com/spf13/cast/compare/e31f36ffc91a...v1.3.0) If you hit the url, you will see the list of changes and the files changed as well. You can also look at who is calling the `spf13/cast` (my favorite is using [rip-grep](https://github.com/BurntSushi/ripgrep)) and you will see that we are not calling it directly from k/k but is used from other code in the vendor directory. It would also be easy to see exactly which methods are being called if you want to go deeper.
> 
> You should also see the other things mentioned here:
> https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/godep.md#reviewing-and-approving-dependency-changes

Sounds good - thanks :)

**Comment by mattjmcnaughton** [comment]
@dims I can't update the original pr but github.com/cyphar/filepath-securejoin v0.2.2 looks good. Its going from 0.2.1+dev to 0.2.2 and has no meaningful changes nor does it add/remove any deps.

**Comment by liggitt** [comment]
See https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/vendor.md#reviewing-and-approving-dependency-changes for review docs (and the rest of that doc for overall approach and goals for bumps)

**Comment by thockin** [comment]
What do the +incompatible mean? Bad semver?

These all look to be tagged versions, but what makes us think this is safe?  For all intents and purposes, this is impossible to review.

**Comment by thockin** [comment]
We really need the bots to warn us about things like new init() functions.

**Comment by BenTheElder** [comment]
> What do the +incompatible mean? Bad semver?

It means go modules doesn't like > 1.0 package versions not following ["semantic import versioning"](https://github.com/golang/go/wiki/Modules#semantic-import-versioning) I.E. it thinks they should be at a different import path. See also [this](https://github.com/golang/go/wiki/Modules#can-a-module-consume-a-v2-package-that-has-not-opted-into-modules-what-does-incompatible-mean).

They all look to still be tagged releases, go just wants > 1.0 to be handled differently.

> We really need the bots to warn us about things like new init() functions.

:+1: 

---

@dims WDYT about PRing one at a time ...?

**Comment by thockin** [comment]
I don't think one PR at a time matters - the commits are clean, but it's
still 30K LOC

File: vendor/github.com/godbus/dbus/transport_generic.go
Comment (BenTheElder) [inline vendor/github.com/godbus/dbus/transport_generic.go]: new init, looks fine

File: vendor/github.com/godbus/dbus/transport_tcp.go
Comment (BenTheElder) [inline vendor/github.com/godbus/dbus/transport_tcp.go]: new init, looks probably fine.

**Review by BenTheElder — APPROVED** [review]
github.com/godbus/dbus looks fine

**Review by BenTheElder — COMMENTED** [review]
github.com/mistifyio/go-zfs changes are trivial, LGTM

File: vendor/github.com/seccomp/libseccomp-golang/seccomp.go
Comment (BenTheElder) [inline vendor/github.com/seccomp/libseccomp-golang/seccomp.go]: ~AFAICT this change is fine, but might be worth noting / keeping an eye on~

This may be a problem https://github.com/bazelbuild/bazel-gazelle/blob/44ce230b3399a5d4472198740358fcd825b0c3c9/language/go/fileinfo.go#L388

File: vendor/github.com/seccomp/libseccomp-golang/BUILD
Comment (BenTheElder) [inline vendor/github.com/seccomp/libseccomp-golang/BUILD]: I think this might actually be a problem (see also review comment below, `seccomp.go` line 23)

https://github.com/bazelbuild/bazel-gazelle/blob/44ce230b3399a5d4472198740358fcd825b0c3c9/language/go/fileinfo.go#L388

https://github.com/bazelbuild/bazel-gazelle/issues/203

**Review by BenTheElder — COMMENTED** [review]
github.com/seccomp/libseccomp-golang might be an issue due to the use of pkg-config instead of LDFLAGS.. I searched locally and don't see any existing packages using this..

**Review by BenTheElder — COMMENTED** [review]
github.com/opencontainers/go-digest LGTM
github.com/cyphar/filepath-securejoin LGTM (trivial diff)
github.com/opencontainers/selinux - could probably use an expert to verify the effects of the logic changes are reasonable ... otherwise LGTM

**Review by BenTheElder — COMMENTED** [review]
github.com/spf13/cast LGTM

**Comment by dims** [comment]
@BenTheElder @mattjmcnaughton @derekwaynecarr - dropped github.com/seccomp/libseccomp from this PR, let's deal with it later

**Comment by dims** [comment]
/assign @thockin 

Tim, since @liggitt is out, can you please `/approve`? a bunch of us have reviewed every commit. 

Took a lot of iterations to get here. Once this lands will file issues for the remaining ones that we could possibly update (one-by-one) and add a white/black list for those that we think we can never get to a tag (example docker/docker does not use a tag). So we can catch this problem of random SHAs earlier during review. This will help us review when new releases are made for the dependencies we can review change logs for performance/security stuff etc which we cannot do today with random SHA(s).

**Comment by thockin** [comment]
Thanks!

/approve

**Review by BenTheElder — APPROVED** [review]
/lgtm
:shipit:

**Comment by dims** [comment]
/hold cancel

Thanks everyone!

**Comment by BenTheElder** [comment]
🎉

**Pull request merged**
